### ISSUE TITLE
no-unsed-vars flags PascalCased type imports even if its used

### ISSUE THREAD

**Issue by Snazzie** [eslint/eslint#13185]
**Tell us about your environment**

* **ESLint Version: 6.8.0**
* **Node Version: v13.9.0**
* **npm Version: 6.13.7**

**What parser (default, Babel-ESLint, etc.) are you using?**
Default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true,
        "node": true
    },
    "extends": [
        "./tsconfig.json",
        "eslint:recommended",
        "plugin:react/recommended",
        "plugin:@typescript-eslint/eslint-recommended"
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": ["react", "@typescript-eslint"],
    "rules": {
        "linebreak-style": ["error", "windows"],
        "quotes": ["error", "double"],
        "semi": ["error", "always"]
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
import { GroupedEntities } from "./types/objectTypes";

export interface CaptureState {
    groupedCaptures: GroupedEntities;
}
```
```ts
export type GroupedEntities = { key: string; values: Entity[] }[];
```

```bash
npm run lint
```

**What did you expect to happen?**
it to not flag Pascal Cased type imports as an error

**What actually happened? Please include the actual, raw output from ESLint.**
it flags any Pascal cased imports as `no-unused-vars` even if its being used.
In this example, it flagged `GroupedEntities`

This doesnt happen for Camel cased  type imports

**Are you willing to submit a pull request to fix this bug?**
no idea where to start.

**Comment by Snazzie** [comment]
turns out my extend tsconfig wasnt valid which probably broke eslint somehow, removed it and did a restart on vscode and error disappeared. still interesting that this behaviour happened though.

**Comment by mdjermanovic** [comment]
Hi @Snazzie, thanks for the issue!

I'm not sure what happened since there should be no difference between camelCased and other variables, but for typescript code you should generally disable the core `no-unused-vars` rule, and use either [`@typescript-eslint/no-unused-vars`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md) or [`@typescript-eslint/no-unused-vars-experimental`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md) instead.

**Comment by Snazzie** [comment]
Thanks for the quick response! i will try that.

**Comment by Snazzie** [comment]
So i am able to suppress the `no-unused-vars` after applying your recommendation.
Another issue has come to light though.

 `no-unused-vars` cant be suppressed on hooks using `// eslint-disable-next-line no-unused-vars`, it can only be suppressed if you suppress the whole file with `/* eslint-disable @typescript-eslint/no-unused-vars */`.

**Comment by Snazzie** [comment]
but i guess thats for another issue.

**Comment by mdjermanovic** [comment]
I think you should completely disable the core `no-unused-vars` rule in your config file, and enable its @typescripts-eslint's equivalent:

```js
"rules": {
    // ...
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error",
    // ...
}
```

(or `@typescript-eslint/no-unused-vars-experimental`)

**Comment by Snazzie** [comment]
that solved the issue. thanks!

**Comment by kaicataldo** [comment]
Closing this issue since the question has been answered.
